python/pyarrow/array.pxi

```diff
@@ -1146,6 +1146,28 @@ cdef class StructArray(Array):
             CStructArray* sarr = <CStructArray*> self.ap
         return pyarrow_wrap_array(sarr.field(ix))
 
+    def field_by_name(self, str name):
```

Comment:
Can't this just be a `.field()` overload? `field_by_name` is a bit verbose.

Reply:
@wesm requested `field_by_name` in the issue https://issues.apache.org/jira/browse/ARROW-3278

Reply:
I'd be fine with overloading `field`. If we do that, we should also do this in `Schema.field` (where there is also `field_by_name`

Reply:
Ok

Reply:
Either that, or raise an exception. We can revise later, IMO.

Reply:
Ok, I'll leave it as is. I'd consider a warning too instead of an exception, should I create a jira?

Reply:
That sounds fine to me.